src/logging.rs

```diff
@@ -0,0 +1,83 @@
+use crate::Logger;
+use crate::Result;
+use wapc_guest::host_call;
+use wascc_codec::logging::*;
+use wascc_codec::serialize;
+
+/// The reserved capability ID for the logging functionality
+pub const CAPID_LOGGING: &str = "wascc:logging";
+
+pub struct DefaultLogger {}
+
+impl DefaultLogger {
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+impl Default for DefaultLogger {
+    fn default() -> Self {
+        DefaultLogger {}
+    }
+}
+
+impl Logger for DefaultLogger {
+    fn log(&self, actor: &str, level: usize, body: &str) -> Result<()> {
```

Comment:
The actors don't know what their own actor ID is. The actor ID is determined by `wascc-host` when it grabs the wapchost ID and looks it up against the list of actors registered in the host. In other words, on the "inside" of this request, the actor ID doesn't need to be a part of the log request, because it'll be given to the capability provider as a separate parameter during the dispatch process. Apologies for not seeing this sooner, as it likely means we need to go back and remove actor from the `WriteLogRequest` struct in the codec.

The API for `log` should be `fn log(&self, level: usize, body: &str)` , and the typed ones should just take a single parameter, e.g. `fn error(&self, body: &str)`